packages/cozy-client/src/store/queries.js

```diff
@@ -148,19 +149,24 @@ const query = (state = queryInitialState, action, documents) => {
         id: action.queryId,
         definition: action.queryDefinition,
         options: action.options,
-        // When the query is new, we set "fetchStatus" to "loading"
-        // directly since we know it will be loaded right away.
-        // This way, the loadQuery action will have no effect, and
-        // we save an additional render.
-        fetchStatus: state.lastUpdate ? state.fetchStatus : 'pending'
+        fetchStatus: state.lastUpdate ? state.fetchStatus : 'pending',
+        isFetching: true
       }
     case LOAD_QUERY:
       if (state.fetchStatus === 'loading') {
         return state
       }
+      if (state.fetchStatus === 'loaded') {
+        return {
+          ...state,
+          fetchStatus: 'loaded',
+          isFetching: true
```

Comment:
the heart of the feature is here: once the query has already been loaded, we don't set fetchStatus to loading.